internal/venice-common/src/main/java/com/linkedin/venice/meta/ReadOnlySchemaRepository.java

```diff
@@ -68,4 +69,29 @@ public interface ReadOnlySchemaRepository extends VeniceResource {
   RmdSchemaEntry getReplicationMetadataSchema(String storeName, int valueSchemaId, int replicationMetadataVersionId);
 
   Collection<RmdSchemaEntry> getReplicationMetadataSchemas(String storeName);
+
+  default DerivedSchemaEntry getLatestDerivedSchema(String storeName) {
+    SchemaEntry valueSchemaEntry = getSupersetOrLatestValueSchema(storeName);
+    try {
+      return getLatestDerivedSchema(storeName, valueSchemaEntry.getId());
+    } catch (Exception e) {
+      /**
+       * Can't find the derived schema for the latest value schema, so it will fall back to find out the latest
+       * value schema, which has a valid derived schema.
+       */
+      Collection<DerivedSchemaEntry> derivedSchemaEntries = getDerivedSchemas(storeName);
```

Comment:
I am a bit confused about this, why don't we do below logic directly? Given that it should always return the known latest one.

Reply:
The latest superset schema might not be the one with the max schema id.

Reply:
Oh...right. This has to be the fallback logic. 

But just wondering if we evolve schema with field reduction, latest value schema and update schema is still not correct right? Even if we use the fall back logic to find a latest derived schema, it might not work? In this case we should still bounce the server to mitigate but SIT will fail still I believe?

Reply:
No, `SIT` will use the returned `DerivedSchemaEntry` to extract the corresponding value schema.